geps/gep-3949/index.md

```diff
@@ -0,0 +1,459 @@
+  in the GatewayClass resource:
+  it allows optionally specifying
+  a reference to a resource
+  that contains configuration
+  specific to the mesh
+  implementation.
+
+### `status`
+
+The `status` stanza
+of the Mesh resource
+is used to indicate
+whether the Mesh resource
+has been accepted by
+a mesh implementation,
+whether the mesh is
+ready to use,
+and
+what features
+the mesh supports.
+
+```yaml
+apiVersion: gateway.networking.x-k8s.io/v1alpha1
+kind: XMesh
+metadata:
+  name: one-mesh-to-mesh-them-all
+  namespace: mesh-namespace
+spec:
+  controllerName: one-mesh.example.com/one-mesh
+status:
+  conditions:
+    # MUST include Accepted and Ready conditions if the Mesh resource is active.
+    - type: Accepted  # Becomes true when the controller accepts the Mesh resource
+      status: "True"
+      reason: Accepted
+      lastTransitionTime: "2023-10-01T12:00:00Z"
+      message: Mesh resource accepted by one-mesh v1.2.3 in namespace one-mesh
+    ...
+  supportedFeatures:
+    # List of SupportedFeature
+    - name: MeshHTTPRoute
+    - name: MeshConsumerRoute
+    - name: OffClusterGateway
+    ...
+```
+
+Although it is
+explicitly
+not a goal of this GEP
+to support multiple meshes
+running in the same cluster at the same time,
+meshes still MUST provide
+human-readable information
+in the `Accepted` condition
+about which mesh instance
+has claimed a given Mesh resource,
+and SHOULD provide human-readable
+information in the `Ready` condition,
+in support of future expansion work.
+This information is meant to be used
+by [Chihiro] and [Ian] as confirmation
+that the mesh instance
+is doing what they expect it to do.
+
+The mesh implementation
+MUST set `status.SupportedFeatures`
+to indicate which features
+the mesh supports.
+
+### Life Cycle
+
+A mesh implementation MUST NOT create a Mesh resource
+if one does not already exist.
+
+If a mesh implementation does not find a Mesh resource
+with a matching `controllerName` at startup:
+
+- It SHOULD warn the user
+  (in whatever way is appropriate for the mesh)
+  that the Mesh resource is missing.
+
+- It MUST act as if a Mesh resource was found
+  with an empty `spec`
+  (other than the `controllerName` field).
+  Optional configuration MUST remain in its default state,
+  and features that require a Mesh resource
+  (such as OCG support)
+  MUST NOT be enabled.
+
+Obviously, if no matching Mesh resource exists,
+the mesh will not be able to publish support features,
+which may lead to assumptions
+that the mesh does not support any features.
+
+Meshes SHOULD provide a default Mesh resource
+when the mesh is installed,
+so that neither [Chihiro] or [Ian] need to know
+the `controllerName`
+before installing the mesh.
+(For example,
+a mesh's Helm chart might include
+a default Mesh resource
+with only the `controllerName` field set,
+with the assumption
+that [Chihiro] or [Ian] will later edit the resource.)
+
+This is in contrast to the GatewayClass resource,
```

Comment:
Not sure this is true. The spec does not say one way or another who creates the GC.
Among `kgateway, istio, envoy-gateway, kong, traefik, cilium, nginx`, only Kong and EnvoyGateway make the user create the GC